micrometer-core/src/main/java/io/micrometer/core/aop/CountedAspect.java

```diff
@@ -0,0 +1,130 @@
+ * annotation and record a few counter metrics about their execution status.
+ *
+ * @author Ali Dehghani
+ * @see Counted
+ */
+@Aspect
+@NonNullApi
+public class CountedAspect {
+
+    /**
+     * The tag name to encapsulate the method execution status.
+     */
+    private static final String RESULT_TAG = "result";
+
+    /**
+     * The tag name to encapsulate the exception thrown by the intercepted method.
+     */
+    private static final String EXCEPTION_TAG = "exception";
+
+    /**
+     * The default metric name.
+     */
+    private static final String DEFAULT_METRIC_NAME = "method.counted";
+
+    /**
+     * Where we're going register metrics.
+     */
+    private final MeterRegistry meterRegistry;
+
+    /**
+     * A function to produce additional tags for any given join point.
+     */
+    private final Function<ProceedingJoinPoint, Iterable<Tag>> tagsBasedOnJoinPoint;
+
+    /**
+     * Construct a new aspect with the given {@code meterRegistry} along with a default
+     * tags provider.
+     *
+     * @param meterRegistry Where we're going register metrics.
+     */
+    public CountedAspect(MeterRegistry meterRegistry) {
+        this(meterRegistry, pjp ->
+                Tags.of("class", pjp.getStaticPart().getSignature().getDeclaringTypeName(),
+                        "method", pjp.getStaticPart().getSignature().getName()));
+    }
+
+    /**
+     * Constructs a new aspect with the given {@code meterRegistry} and tags provider function.
+     *
+     * @param meterRegistry        Where we're going register metrics.
+     * @param tagsBasedOnJoinPoint A function to generate tags given a join point.
+     */
+    public CountedAspect(MeterRegistry meterRegistry, Function<ProceedingJoinPoint, Iterable<Tag>> tagsBasedOnJoinPoint) {
+        this.meterRegistry = meterRegistry;
+        this.tagsBasedOnJoinPoint = tagsBasedOnJoinPoint;
+    }
+
+    /**
+     * Intercept methods annotated with the {@link Counted} annotation and expose a few counters about
+     * their execution status. By default, this aspect records both failed and successful attempts. If the
+     * {@link Counted#recordFailuresOnly()} is set to {@code true}, then the aspect would record only
+     * failed attempts. In case of a failure, the aspect tags the counter with the simple name of the thrown
+     * exception.
+     *
+     * @param pjp     Encapsulates some information about the intercepted area.
+     * @param counted The annotation.
+     * @return Whatever the intercepted method returns.
+     * @throws Throwable When the intercepted method throws one.
+     */
+    @Around("@annotation(counted)")
+    public Object interceptAndRecord(ProceedingJoinPoint pjp, Counted counted) throws Throwable {
+        try {
+            Object result = pjp.proceed();
+            if (!counted.recordFailuresOnly()) {
+                record(pjp, counted, "None", "success");
```

Comment:
Looks like the `@TimedAspect` used a lowercase `none`

Reply:
✅

Reply:
It is unfortunate that `WebMvcTags` in the Spring support uses `None`, though. We should probably align these in 2.0.